.changeset/many-rivers-deny.md

```diff
@@ -0,0 +1,5 @@
+---
+'@primer/react': minor
+---
+
+Adds new sub-component to `NavList`, `NavList.Expand`
```

Comment:
```suggestion
Add `NavList.ShowMoreItem` component to support showing more content within a `NavList`.
```

Just an idea for the changelog, really just `NavList.Expand` -> `NavList.ShowMoreItem` is all that's needed.